calm_llm/data/flows/book_hotel.yml

```diff
@@ -4,7 +4,16 @@ flows:
     steps:
       - collect: trip_destination
         reset_after_flow_ends: false      
-      - collect: hotel_price_range
+      - id: "verify_hotel_location_is_in_db"
```

Comment:
wouldn't it be simpler to use a rejection? https://rasa.com/docs/rasa-pro/building-assistants/validating-slot-values#validating-a-slot-values-format-with-a-condition

Reply:
Yeah, good question. I thought about that but decided that the current approach follows best practices, for following reasons:

- I considered a `rejection` but the check is too difficult to do it easily via pyred. It needs to handle conversion to lowercase and then check against a range of values. Is probably possible, but I think it will be hard to maintain and debug.

- I considered a slot_validation with a custom action on trip_destination, but that would hide the logic inside a custom action. The trip_destination slot is a global slot, and to check if it is in the database as a hotel location needs to run only for the search_hotel flow. We should not be hiding that type of logic inside a custom action.

Reply:
> We should not be hiding that type of logic inside a custom action

Agreed! This also makes it easier to work in Studio, because the logic is in the flow